static/src/js/containers/DownloadHistoryContainer/__tests__/DownloadHistoryContainer.test.js

```diff
@@ -1,89 +1,368 @@
+      test('renders a message', () => {
+        setup({
+          earthdataEnvironment: 'prod',
+          retrievalHistory: [],
+          retrievalHistoryLoading: false,
+          retrievalHistoryLoaded: true,
+          onDeleteRetrieval: jest.fn()
+        })
+
+        expect(screen.queryByRole('table')).not.toBeInTheDocument()
+        expect(screen.queryByTestId('spinner')).not.toBeInTheDocument()
+        expect(screen.getByText('No download history to display.')).toBeInTheDocument()
+      })
+    })
+
+    describe('when a retrieval exists with one collection that has no title', () => {
+      test('renders a table with collection count', () => {
+        setup({
+          earthdataEnvironment: 'prod',
+          retrievalHistory: [{
+            id: '8069076',
+            jsondata: {},
+            created_at: '2019-08-25T11:58:14.390Z',
+            collections: [{}]
+          }],
+          retrievalHistoryLoading: false,
+          retrievalHistoryLoaded: true,
+          onDeleteRetrieval: jest.fn()
+        })
+
+        expect(screen.getByRole('table')).toBeInTheDocument()
+        expect(screen.getByRole('link', { name: '1 collection' })).toHaveAttribute('href', '/downloads/8069076')
+      })
+    })
+
+    describe('when a retrieval exists with one collection', () => {
+      test('renders a table with the collection title', () => {
+        setup({
+          earthdataEnvironment: 'prod',
+          retrievalHistory: [{
+            id: '8069076',
+            jsondata: {},
+            created_at: '2019-08-25T11:58:14.390Z',
+            collections: [{
+              title: 'Collection Title'
+            }]
+          }],
+          retrievalHistoryLoading: false,
+          retrievalHistoryLoaded: true,
+          onDeleteRetrieval: jest.fn()
+        })
+
+        expect(screen.getByRole('table')).toBeInTheDocument()
+        expect(screen.getByRole('link', { name: 'Collection Title' })).toHaveAttribute('href', '/downloads/8069076')
+      })
+    })
+
+    describe('when a retrieval exists with two collections', () => {
+      test('renders a table with the first collection title and count of others', () => {
+        setup({
+          earthdataEnvironment: 'prod',
+          retrievalHistory: [{
+            id: '8069076',
+            jsondata: {},
+            created_at: '2019-08-25T11:58:14.390Z',
+            collections: [{
+              title: 'Collection Title'
+            }, {
+              title: 'Collection Title Two'
+            }]
+          }],
+          retrievalHistoryLoading: false,
+          retrievalHistoryLoaded: true,
+          onDeleteRetrieval: jest.fn()
+        })
+
+        expect(screen.getByRole('table')).toBeInTheDocument()
+        expect(screen.getByRole('link', { name: 'Collection Title and 1 other collection' })).toHaveAttribute('href', '/downloads/8069076')
+      })
+    })
+
+    describe('when portals were used to place an order', () => {
```

Comment:
I made this comment earlier, but none of the logic described in this test is part of DownloadHistoryContainer, and should be tested within PortalLinkContainer, this test should be removed.